core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -530,18 +530,29 @@ sealed abstract class IO[+E, +A] extends Serializable { self =>
    * Times out this action by the specified duration.
    *
    * {{{
-   * IO.point(1).timeout(Option.empty[Int])(Some(_))(1.second)
+   * IO.point(1).timeout0(Option.empty[Int])(Some(_))(1.second)
    * }}}
    */
-  final def timeout[B](z: B)(f: A => B)(duration: Duration): IO[E, B] =
+  final def timeout0[Z](z: Z)(f: A => Z)(duration: Duration): IO[E, Z] =
```

Comment:
private？

Reply:
This is kind of a pattern where the `-0` suffix identifies the more powerful version of a combinator: `fork0`, `bracket0`, `repeatOrElse0`, etc. The idea is all other combinators can be/are derived from it. Most users will be happy with the derived ones, but some may have particular needs only the `-0` version allows.